code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -330,6 +368,34 @@ func (sc *StorageSmartContract) verifyChallenge(t *transaction.Transaction,
 	input []byte, balances c_state.StateContextI) (resp string, err error) {
 
 	var challResp ChallengeResponse
+
+	conf, err := sc.getConfig(balances, true)
+	if err != nil {
+		return "", common.NewError("verify_challenge",
+			"cannot get smart contract configurations: "+err.Error())
+	}
+
+	if balances.GetBlock().Round%conf.BlockReward.ChallengePeriod == 0 {
+		// clean active_passed_blobbers partition
+		_, err = balances.DeleteTrieNode(ACTIVE_PASSED_BLOBBERS_KEY)
+		if err != nil {
+			return "", common.NewError("verify_challenge",
+				"cannot delete active partition: "+err.Error())
+		}
+
+		// move ongoing_passed_blobbers to active_passed_blobbers
+		ongoingList, err := getOngoingPassedBlobbersList(balances)
+		if err != nil {
+			return "", common.NewError("verify_challenge",
+				"cannot get ongoing partition: "+err.Error())
+		}
+		err = ongoingList.Migrate(ACTIVE_PASSED_BLOBBERS_KEY, balances)
```

Comment:
I am not sure what Migrate does but it looks far too complicated. Can't you just do
```go
_, err = balances.InsertTrieNode(ACTIVE_PASSED_BLOBBERS_KEY, ongoingList)
_, err = balances.InsertTrieNode(ONGOING_PASSED_BLOBBERS_KEY, emptyPartition_plus_one_pass_from_this_call)
```

Reply:
@Sriep what about the list stored in ongoingList, they also needs to be migrated no?
else the keys will start clashing in future rounds
Partition.Key()

Reply:
superb. let me do that